summarization/stamp_page_picking/interesting_sequence_picker.py

```diff
@@ -0,0 +1,233 @@
+            stamp_pages,
+            summary_sentences,
+            max_pages_allowed):
+        self.stamp_pages = stamp_pages
+        self.summary_sentence_embeddings = [
+            sentence.embedding for sentence in
+            summary_sentences
+        ]
+        self.max_pages_allowed = max_pages_allowed
+
+        self.stamp_page_indices = list(range(len(self.stamp_pages)))
+
+        self.cover_size = len(summary_sentences)
+        self.picked_cover = [0] * self.cover_size
+
+        # the first stamp page is the title card
+        # for the series of stamp pages
+        # we will use that as the seed stamp pages
+        self.last_picked_stamp_page_index = self.stamp_pages[0]
+        self._pick_cover_at_index(0)
+        self.stamp_page_indices.pop(0)
+
+        self.threshold = 0.5  # change as required
+
+        # the final sequence that will be returned
+        self.stamp_page_sequence = list()
+
+        # unit weight as of now
+        # can be changed as per requirement
+        self.individual_score_weights = [
+            1,  # content change score weight
+            1,  # sentiment change score
+            1,  # textual entailment score
+            1,  # unpicked weights score
+            1   # sweeping index score
+        ]
+
+        # sweeping index is initially the maximum or last index possible
+        self.sweeping_index = max(
+            [max(page.media_index, page.sentence_index)
+             for page in self.stamp_pages]
+        )
+        # step size to change sweeping index
+        # TODO: add variable change in step size
+        # based on iteration
+        self.step_size = 1
+
+    def get_interesting_sequence(self):
+        for i in range(self.max_pages_allowed):
+            # get the index of the next best stamp page
+            stamp_page_index = self._get_next_best_stamp_page_index()
+
+            # do not continue if no more stamp pages
+            # are left to be picked
+            if stamp_page_index == -1:
+                break
+
+            # append the stamp page at that index
+            self.stamp_page_sequence.append(
+                self.stamp_pages[stamp_page_index]
+            )
+
+            # update the cover for unpicked weights
+            # calculation during next iteration
+            self._pick_cover_at_index(stamp_page_index)
+
+            # update the sweeping index
+            self._update_sweeping_index()
+
+            # update last picked
+            self.last_picked_stamp_page_index = stamp_page_index
+
+        return self.stamp_page_sequence
+
+    def _set_stamp_page_covers(self):
+        ''' find covers for stamp pages'''
+        preprocessor = BudgetedMaxCoverPreprocessor(
+            self.stamp_pages,
+            self.summary_sentence_embeddings,
+            self.threshold)
+        self.stamp_page_covers \
+            = preprocessor.get_cover_objects_for_stamp_pages()
+
+    def _pick_cover_at_index(self, index):
```

Comment:
Could this be named better, since we are adding the picked stamp page's cover here keep a track of the total content covered from the page.

Reply:
Updated accordingly!